pkg/controllers/settings/settings.go

```diff
@@ -0,0 +1,117 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package settings
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/avast/retry-go"
+	"github.com/samber/lo"
+	"go.uber.org/multierr"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/system"
+
+	"github.com/aws/karpenter-core/pkg/utils/atomic"
+)
+
+const (
+	controllerName = "global-settings"
+	configMapName  = "karpenter-global-settings"
+)
+
+type keyType struct{}
+
+var Key = keyType{}
+
+type Interface interface {
+	Get() map[string]string
+	Update(map[string]string)
+	Inject(context.Context) context.Context
+	RegisterValidator(validatorFunc Validator)
+}
+
+type Validator func(map[string]string) error
+
+// Settings stores a thread-safe version of the "karpenter-global-settings"
+type Settings struct {
+	mu sync.RWMutex
+
+	validators atomic.Slice[Validator]
+	data       map[string]string
+}
+
+func NewSettings(ctx context.Context, clientset *kubernetes.Clientset) *Settings {
+	data := map[string]string{}
+
+	// Perform initial settings hydration for components that rely on it in the constructor
+	// If we don't succeed at doing the hydration, we just leave it alone, and we will take the default later
+	_ = retry.Do(func() error {
+		cm, err := clientset.CoreV1().ConfigMaps(system.Namespace()).Get(ctx, configMapName, metav1.GetOptions{})
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return nil
+			}
+			return err
+		}
+		data = lo.Assign(cm.Data)
+		return err
+	})
+	logging.FromContext(ctx).Debugf("Hydrated global settings")
```

Comment:
nit: they may already be hydrated.
```suggestion
	logging.FromContext(ctx).Debugf("Refreshed global settings")
```